test/e2e/annotations/satisfy.go

```diff
@@ -0,0 +1,117 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package annotations
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/parnurzeal/gorequest"
+	"k8s.io/ingress-nginx/test/e2e/framework"
+)
+
+var _ = framework.IngressNginxDescribe("Annotations - SATISFY", func() {
+	f := framework.NewDefaultFramework("satisfy")
+
+	BeforeEach(func() {
+		f.NewEchoDeployment()
+	})
+
+	AfterEach(func() {
+	})
+
+	It("should set satisfy any in nginx configuration file", func() {
+		host := "satisfy"
+
+		annotations := map[string]string{
+			"nginx.ingress.kubernetes.io/satisfy": "any",
+		}
+
+		ing := framework.NewSingleIngress(host, "/", host, f.IngressController.Namespace, "http-svc", 80, &annotations)
+		f.EnsureIngress(ing)
+
+		f.WaitForNginxServer(host,
+			func(server string) bool {
+				return Expect(server).Should(ContainSubstring("satisfy any"))
+			})
+
+		resp, body, errs := gorequest.New().
+			Get(f.IngressController.HTTPURL).
+			Retry(10, 1*time.Second, http.StatusNotFound).
+			Set("Host", host).
+			End()
+
+		Expect(errs).Should(BeEmpty())
+		Expect(resp.StatusCode).Should(Equal(http.StatusOK))
+		Expect(body).Should(ContainSubstring(fmt.Sprintf("host=%v", host)))
+	})
+
+	It("should set satisfy all in nginx configuration file", func() {
+		host := "satisfy"
+
+		annotations := map[string]string{
+			"nginx.ingress.kubernetes.io/satisfy": "all",
+		}
+
+		ing := framework.NewSingleIngress(host, "/", host, f.IngressController.Namespace, "http-svc", 80, &annotations)
+		f.EnsureIngress(ing)
+
+		f.WaitForNginxServer(host,
+			func(server string) bool {
+				return Expect(server).Should(ContainSubstring("satisfy all"))
+			})
+
+		resp, body, errs := gorequest.New().
+			Get(f.IngressController.HTTPURL).
+			Retry(10, 1*time.Second, http.StatusNotFound).
+			Set("Host", host).
+			End()
+
+		Expect(errs).Should(BeEmpty())
+		Expect(resp.StatusCode).Should(Equal(http.StatusOK))
+		Expect(body).Should(ContainSubstring(fmt.Sprintf("host=%v", host)))
+	})
+
+	It("should set satisfy all for incorrect satisfy value", func() {
+		host := "satisfy"
+
+		annotations := map[string]string{
+			"nginx.ingress.kubernetes.io/satisfy": "invalid",
+		}
+
+		ing := framework.NewSingleIngress(host, "/", host, f.IngressController.Namespace, "http-svc", 80, &annotations)
+		f.EnsureIngress(ing)
+
+		f.WaitForNginxServer(host,
+			func(server string) bool {
+				return Expect(server).Should(ContainSubstring("satisfy all"))
+			})
+
+		resp, body, errs := gorequest.New().
+			Get(f.IngressController.HTTPURL).
+			Retry(10, 1*time.Second, http.StatusNotFound).
+			Set("Host", host).
+			End()
+
+		Expect(errs).Should(BeEmpty())
+		Expect(resp.StatusCode).Should(Equal(http.StatusOK))
+		Expect(body).Should(ContainSubstring(fmt.Sprintf("host=%v", host)))
+	})
```

Comment:
Please avoid creating different `It` blocks for each case that has everything same except the annotation value and single assertion. Instead I'd suggest you create only single `It` something like

```
It("configures satisfy directive correctly", func() {
  // create your ingress
  // in a loop iterate through every test case, update ingress annotation and do the assertion
})
```

this will reduce duplication significantly and will still be clear enough. Also it'll run faster because you won't be deploying and tearing down ingress-nginx 3 times (before running an It block we start with a fresh namespace and ingress-nginx deplotment).

Reply:
Done